src/kafka-util/src/client.rs

```diff
@@ -280,8 +308,80 @@ where
     const ENABLE_REFRESH_OAUTH_TOKEN: bool = C::ENABLE_REFRESH_OAUTH_TOKEN;
 
     fn rewrite_broker_addr(&self, addr: BrokerAddr) -> BrokerAddr {
-        match self.rewrites.get(&addr) {
-            None => addr,
+        let mut rewrites = self.rewrites.blocking_lock();
+
+        match rewrites.get(&addr) {
+            None => {
+                match &self.default_override {
+                    Some(default_tunnel) => {
+                        let broker = addr.clone();
+                        let (tunnel_id, default_tunnel) = default_tunnel.clone();
+                        let ssh_tunnel_manager = self.ssh_tunnel_manager.clone();
+                        let runtime = self.runtime.clone();
+                        // We spawn an additional thread here, and immediately
+                        // join it so that the rdkafka thread calling this callback
+                        // spends all its time parked, instead of being
+                        // continuously woken up when we `block_on` the
+                        // ssh tunnel creation.
```

Comment:
I'm still confused about why we need to do this. What goes wrong if the rdkafka thread is woken up repeatedly?

Reply:
I just dont really trust the rdkafka thread to be normal, so im just being defensive!